sale_discount_total/models/discount_approval.py

```diff
@@ -37,24 +38,30 @@ class sale_discount(models.Model):
 
     @api.multi
     def action_confirm(self):
+        if self._get_forbidden_state_confirm() & set(self.mapped('state')):
+            raise UserError(_("It is not allowed to confirm an order in the following states: %s")
+                            % (", ".join(self._get_forbidden_state_confirm())))
         discnt = 0.0
         no_line = 0.0
         for order in self:
-            if self.company_id.so_double_validation == 'two_step':
+            if order.company_id.so_double_validation == 'two_step':
```

Comment:
Please, review this change...
Personally, I think that it is OK, if you are calling `action_confirm` on a recordset of multiple Sales Order... But I'm not sure of it.

Thank you.